packages/big-design/src/components/TableNext/types.ts

```diff
@@ -12,8 +12,11 @@ export interface TableSelectable {
 
 export interface TableExpandable<T> {
   expandedRows: Record<string, true>;
-  onExpandedChange(expandedItems: Record<string, true>): void;
   expandedRowSelector: (item: T) => T[] | undefined;
+  onExpandedChange(expandedItems: Record<string, true>, expandedIndex: number): void;
+  renderHelperRow?:
+    | ComponentType<{ parentRowIndex: number; children?: ReactNode }>
```

Comment:
are we planning to have different implementations of the helper row because i'm not sure why the type is a renderer. Also, should probably rename it to helperRowRenderer because at first thought, it seems like a boolean property

Reply:
Good point, I think it's more descriptive 👍🏼 . I'll update it

Reply:
ohh this is also gonna help render the button, or the loading icon based on the status of each parent row. 

Example: 
```tsx
  const HelperRow = ({ parentRowIndex }) => {
  
    if (state[parentIndex].isLoading) {
       return <LoadingIcon/>
    }

   if (state[parentIndex] !== undefined && state[parentIndex].remaining) {
    const itemsToView = state[parentIndex].remaining;
     return <Button onClick={viewMore(parentRowIndex)} />
   }

   return null
  };
```